lib/puppet/provider/firewalld.rb

```diff
@@ -101,7 +102,6 @@ def online?
   end
 
   def self.online?
-    check_running_state unless state == true
```

Comment:
This needs to run each time.  If `state` was `false` (instead of `nil`) it needs to be rechecked.